e2e/containers.go

```diff
@@ -95,71 +98,52 @@ func createHTTPServerContainer(alias string, tmpDir *helpertest.TmpFolder,
 	})
 }
 
-func createRedisContainer() (testcontainers.Container, error) {
-	ctx := context.Background()
-
-	req := testcontainers.ContainerRequest{
-		Image:          redisImage,
-		Networks:       []string{NetworkName},
-		ExposedPorts:   []string{"6379/tcp"},
-		NetworkAliases: map[string][]string{NetworkName: {"redis"}},
-		WaitingFor:     wait.ForExposedPort(),
+func WithNetwork(network string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.NetworkAliases = map[string][]string{NetworkName: {network}}
+		req.Networks = []string{NetworkName}
 	}
+}
 
-	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
-		ContainerRequest: req,
-		Started:          true,
-	})
+func createRedisContainer() (*redis.RedisContainer, error) {
+	ctx := context.Background()
 
-	return container, err
+	return redis.RunContainer(ctx,
+		testcontainers.WithImage(redisImage),
+		redis.WithLogLevel(redis.LogLevelVerbose),
+		WithNetwork("redis"),
+	)
 }
 
-func createPostgresContainer() (testcontainers.Container, error) {
+func createPostgresContainer() (*postgres.PostgresContainer, error) {
 	ctx := context.Background()
 
-	req := testcontainers.ContainerRequest{
-		Image:          postgresImage,
-		Networks:       []string{NetworkName},
-		ExposedPorts:   []string{"5432/tcp"},
-		NetworkAliases: map[string][]string{NetworkName: {"postgres"}},
-		Env: map[string]string{
-			"POSTGRES_USER":     "user",
-			"POSTGRES_PASSWORD": "user",
-		},
-		WaitingFor: wait.ForExposedPort(),
-	}
+	const waitLogOccurrence = 2
 
-	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
-		ContainerRequest: req,
-		Started:          true,
-	})
+	return postgres.RunContainer(ctx,
+		testcontainers.WithImage(postgresImage),
 
-	return container, err
+		postgres.WithDatabase("user"),
+		postgres.WithUsername("user"),
+		postgres.WithPassword("user"),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").
+				WithOccurrence(waitLogOccurrence).
+				WithStartupTimeout(startupTimeout)),
```

Comment:
I was wondering why this is the only one using `WithWaitStrategy` and from looking at `testcontainers` code, the others have one built-in.  
Might be worth making an issue/PR to add a built-in one for postgres too since the others also use log matching, so there's no reason this couldn't be built-in.

Reply:
Yes, I have also noticed that. In official example for postgres (https://golang.testcontainers.org/modules/postgres/) is a wait strategy defined. In redis and mariaDB not.

Reply:
Found this comment: https://github.com/testcontainers/testcontainers-go/issues/1619#issuecomment-1719278332

Reply:
Nice find, glad they're thinking of it.